### ISSUE TITLE
Type generation with Typescript 3.6.x broken

### ISSUE THREAD

**Issue by valentinpalkovic** [prisma-labs/nexus#226]
Nexus version: 0.12.0-beta-9
Typescript version: 3.6.3

Example Schema:
```
type Icon {
  content: String!
  height: Float!
  id: ID!
  width: Float!
}


type Logo {
  icon: Icon!
}
```

Generated output:
```
import { objectType } from 'nexus';

const Icon = objectType({
  name: 'Icon',
  definition(t) {
    t.string('content');
    t.float('height');
    t.id('id');
    t.float('width');
  },
});
const Logo = objectType({
  name: 'Logo',
  definition(t) {
    t.field('icon', { type: Icon });
  },
});
```

Error at `t.field('icon', { type: Icon })`:
```
Type 'NexusObjectTypeDef<"Icon">' is not assignable to type '"Icon" | "Query" | "Boolean" | "Float" | "ID" | "Int" | "String" | NexusInterfaceTypeDef<string> | NexusObjectTypeDef<string> | NexusUnionTypeDef<string> | NexusEnumTypeDef<string> | NexusScalarTypeDef<...> | NexusWrappedType<...>'.
  Type 'NexusObjectTypeDef<"Icon">' is not assignable to type 'NexusObjectTypeDef<string>'.
    Types of property 'config' are incompatible.
      Type 'NexusObjectTypeConfig<"Icon">' is not assignable to type 'NexusObjectTypeConfig<string>'.
        Types of property 'definition' are incompatible.
          Type '(t: ObjectDefinitionBlock<"Icon">) => void' is not assignable to type '(t: ObjectDefinitionBlock<string>) => void'.
            Types of parameters 't' and 't' are incompatible.
              Type 'ObjectDefinitionBlock<string>' is not assignable to type 'ObjectDefinitionBlock<"Icon">'.
                Types of property 'typeBuilder' are incompatible.
                  Type 'ObjectDefinitionBuilder<string>' is not assignable to type 'ObjectDefinitionBuilder<"Icon">'.
                    Types of property 'addFieldModifications' are incompatible.
                      Type '<FieldName extends string>(changes: FieldModificationDef<string, FieldName>) => void' is not assignable to type '<FieldName extends string>(changes: FieldModificationDef<"Icon", FieldName>) => void'.
                        Types of parameters 'changes' and 'changes' are incompatible.
                          Type 'FieldModificationDef<"Icon", any>' is not assignable to type 'FieldModificationDef<string, any>'.
                            Types of property 'resolve' are incompatible.
                              Type 'FieldResolver<"Icon", any> | undefined' is not assignable to type 'FieldResolver<string, any> | undefined'.
                                Type 'FieldResolver<"Icon", any>' is not assignable to type 'FieldResolver<string, any>'.
                                  Type 'string' is not assignable to type '"Icon"'.
```

tsconfig file:
```
{
  "compilerOptions": {
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "noUnusedLocals": true,
    "rootDir": "src",
    "module": "commonjs",
    "target": "es6",
    "sourceMap": true,
    "outDir": "dist",
    "baseUrl": ".",
    "jsx": "react",
    "paths": {
      "*": ["node_modules/*", "src/types/*"],
      "@logo-tool/shared": ["../shared/build"]
    }
  },
  "include": ["src/**/*.ts"]
}
```

**Comment by jasonkuhrt** [comment]
Hey @valentinpalkovic I don't think this issue is related to TS 3.6. We've added support for that in #255 and works without changes that would seem related to your issue.

Can you please share a repro for us to better understand?

**Comment by tgriesser** [comment]
I think I'm able to reproduce this: https://circleci.com/gh/prisma-labs/nexus/1363

**Comment by valentinpalkovic** [comment]
@jasonkuhrt Repository is here available:
https://github.com/valentinpalkovic/nexus-0.12.0-beta.13-typescript-3.6.4

1. `yarn`
2. `yarn start`

Now you should already see an error in vscode in line https://github.com/valentinpalkovic/nexus-0.12.0-beta.13-typescript-3.6.4/blob/master/src/index.ts#L17, after `schema.graphql` and `typings.ts` were generated. Also `yarn build` will fail afterwards.

**Comment by valentinpalkovic** [comment]
Fixed with the lastest release 0.12.0-beta.13.
